Frame unification

guillep opened the pull request:
Slight differences in the shape of frames between the pure interpreter and the VM with JIT make a lot of code duplication and make it difficult to change and debug.

This PR unifies both. The main idea that the JIT VM keeps the same frame organisation, it is the non-JIT VM (used to debug mainly nowadays) that accommodates slightly.

The core differences dealt with are:
 - the interpreter frames in the JIT'ted VM have an extra field to store its instruction pointer when calling a machine code method. Instead of duplicating the entire code, we conditionally push/iterate over that extra field in one place.
 - the base frames in the stack need to store a trampoline address in the caller saved IP to return gracefully if it is a machine code frame, and it then stores the caller context in a separate field at the base of the stack => the non-JIT VM now adopts the same organization, except that it never uses a trampoline to return

Other minor details include the refactoring in the management of instruction pointers which in the JITt'ed VM can have a mixture of machine code and interpreter instruction pointers. The code should do the same but just avoid repetition.

Also, several minor cleanups of dead code observed on the way

syrel committed:
remove inline from worker_dispatch_callout

guillep committed:
Added test on extended store and pop

privat commented on smalltalksrc/VMMaker/Cogit.class.st:
```diff
@@ -266,7 +255,9 @@ Class {
-		'jitCodeZoneWriteEnabled'
+		'jitCodeZoneWriteEnabled',
+		'cePrimReturnEnterCogCode',
+		'cePrimReturnEnterCogCodeProfiling'
```
why those two ivars were moved around?

guillep replied:
no idea :)

privat commented on smalltalksrc/VMMaker/InterpreterPrimitives.class.st:
```diff
@@ -325,15 +319,12 @@ InterpreterPrimitives >> isInstanceOfClassArray: oop [
-	"N.B.  Because Slang always inlines is:instanceOf:compactClassIndex:
-	 (because is:instanceOf:compactClassIndex: has an inline: pragma) the
-	 phrase (objectMemory splObj: ClassBlockClosure) is expanded in-place
-	 and is _not_ evaluated if oop has a non-zero CompactClassIndex."
```
is the comment now false?

guillep replied:
Since a long time. There is no call to `(objectMemory splObj: ClassBlockClosure)` in this method :)

privat commented on smalltalksrc/VMMaker/SpurMemoryManager.class.st:
```diff
@@ -878,9 +878,6 @@ SpurMemoryManager class >> initializeSpecialObjectIndices [
 	SelectorCannotInterpret := 34.
-	"Was MethodContextProto := 35."
-	"Was ClassBlockClosure := 36."
-	"Was BlockContextProto := 37."
```
do you plan to overwrite them? Could this cause issues with old images?

guillep replied:
> do you plan to overwrite them?
There are no plans, but those three indexes have not been used for the last couple of years. 

> Could this cause issues with old images?

Our VM does not run veeery old images.

privat commented on smalltalksrc/VMMaker/SpurMemoryManager.class.st:
```diff
@@ -9050,8 +9047,8 @@ SpurMemoryManager >> methodHeaderFromSavedFirstField: oop [
 { #category : #'memory access' }
 SpurMemoryManager >> methodHeaderOf: methodObj [
 	"Answer the method header of a CompiledMethod object."
-	self assert: (self isCompiledMethod: methodObj).
```
why remove assert?

guillep replied:
Because asserts and inlines don't play well together right now :(

privat commented on smalltalksrc/VMMaker/StackInterpreter.class.st:
```diff
@@ -891,49 +891,57 @@ StackInterpreter class >> initializeFrameIndices [
-			context (uninitialized)
+			context (initialized to nil)
+			frame flags (interpreter only)
+			saved method ip (initialized to 0; only in interpreter when VM supports JIT compilation)
```
"only in" or "only used in"?

guillep replied:
only in. If the VM does not have JIT, that field is not even added at all.

privat commented on smalltalksrc/VMMaker/StackInterpreter.class.st:
```diff
@@ -8717,6 +8762,14 @@ StackInterpreter >> log10: n [
 ]
 
+{ #category : #logging }
+StackInterpreter >> logDebug: aString [
```
only 3 senders. logging infra should be factorized with `--logLevel=` (in another PR)

privat commented on smalltalksrc/VMMaker/StackInterpreter.class.st:
```diff
@@ -9071,102 +9124,126 @@ StackInterpreter >> lookupSelector: selector inClass: class [
+	"We finished pushing all the meta-data required for the base-frame to work.
+	Proceed to start a new frame"
+	"Create either a machine code frame or an interpreter frame based on the pc.  If the pc is -ve
+	 it is a machine code pc and so we produce a machine code frame.  If +ve an interpreter frame.
```
what is `ve`?

guillep replied:
See my other comment!

privat commented on smalltalksrc/VMMaker/StackInterpreter.class.st:
```diff
@@ -10264,7 +10323,18 @@ StackInterpreter >> moveFramesIn: oldPage through: theFP toPage: newPage [
+
+	"In cog the base frames are married, in non cog they are not"
+	self isCog ifTrue: [
+		theContext := self ensureFrameIsMarried: theFP SP: (self frameReceiverLocation: theFP).
```
`isCog` here feels inconsistent with the rest of the PR

What about a method that returns nil in the StackInterpretor and possibly perform matrimonial work in the CoInterpretor?

moreover, the method `moveFramesIn...` is very similar that the one overridden in CoInterpretor, so I'm not even sure that the isCog is meaningful in neither. Could it be possible to have a single method `moveFramesIn...` for both?

guillep replied:
Yes, I refactored so they look very similar (and the repetition could eventually be removed). But at some point I put my time on fixing issue and making the PR pass ^^

privat commented on smalltalksrc/VMMaker/CoInterpreter.class.st:
```diff
@@ -2157,17 +2073,20 @@ CoInterpreter >> contextInstructionPointer: theIP frame: theFP [
+	theIP := theIP = cogit ceReturnToInterpreterPC
+		         ifTrue: [ self iframeSavedIP: theFP ]
+		         ifFalse: [ theIP ].
+	^ super contextInstructionPointer: theIP frame: theFP
```
I'm waiting for my chocolate

privat commented on smalltalksrc/VMMaker/CoInterpreter.class.st:
```diff
@@ -3348,6 +3113,19 @@ CoInterpreter >> instructionPointerAddress [
+	self assert: HasBeenReturnedFromMCPC < 0.
+	^ (objectMemory isIntegerObject: theIP)
+		  ifTrue: [ objectMemory integerValueOf: theIP ]
+		  ifFalse: [ HasBeenReturnedFromMCPC ]
```
what is HasBeenReturnedFromMCPC? How could theIP be not an integer?

guillep replied:
it can be nil if the context already returned!

privat commented on smalltalksrc/VMMaker/CoInterpreter.class.st:
```diff
@@ -4401,10 +3815,18 @@ CoInterpreter >> moveFramesIn: oldPage through: theFP toPage: newPage [
+		theContext := self ensureFrameIsMarried: theFP SP: (self frameReceiverLocation: theFP).
+	] ifFalse: [
+		theContext := objectMemory nilObject.
+	].
```
see other remark about `isCog` and `moveFramesIn:...`

privat reviewed (commented):
The PR is big thus very hard to read, but the work is impressive. I tried my best to review it even if lot of things are still mysterious to me.

guillep commented on smalltalksrc/VMMaker/StackInterpreter.class.st:
```diff
@@ -9071,102 +9124,126 @@ StackInterpreter >> lookupSelector: selector inClass: class [
+
+	"We finished pushing all the meta-data required for the base-frame to work.
+	Proceed to start a new frame"
+	"Create either a machine code frame or an interpreter frame based on the pc.  If the pc is -ve
+	 it is a machine code pc and so we produce a machine code frame.  If +ve an interpreter frame.
```
```suggestion
	"Create either a machine code frame or an interpreter frame based on the pc.  If the pc is negative
	 it is a machine code pc and so we produce a machine code frame.  If positive an interpreter frame.
```

PalumboN committed:
Hi primitiveFormat!

PalumboN committed:
Ups, put primitiveFormat in primitive table

tesonep committed:
Merge pull request #746 from pharo-project/preparing-release-v10.1.0

Preparing release v10.1.0

guillep committed:
Merge branch 'pharo-10' into fix-vmmaker-command-2

tesonep committed:
Merge pull request #741 from Rinzwind/fix-vmmaker-command-2

Change custom command in ‘vmmaker.cmake’ to take into account that the ‘CMAKE_CURRENT_BINARY_DIR_TO_OUT’ can contain other characters besides spaces that require escaping

tesonep committed:
Merge pull request #709 from pharo-project/backport-pr-new-old

Backporting PR for allocating in the old space

Rinzwind committed:
Added comment to ‘tty_spawn’ describing its functionality and the motivation for providing it through a library included with the VM.

Rinzwind committed:
Added definition for ‘_XOPEN_SOURCE’ to ‘tty.c’.

Rinzwind committed:
Merged commit 55f077d787401be7 (latest ‘pharo-10’).

Rinzwind committed:
Made ‘tty_spawn’ also check for errors when opening the slave pseudo-terminal device and setting up the standard file descriptors.

Rinzwind committed:
Changed exit status for errors in the setup of the child process in ‘tty_spawn’ from 1 to 127 (similar to ‘posix_spawn’).

tesonep committed:
Adding generation of signature files

tesonep committed:
Fixing memory map in OSX

tesonep committed:
Merge pull request #751 from tesonep/fixing-memory-map-in-osx

Fixing memory map in OSX

tesonep committed:
Merge pull request #749 from tesonep/adding-signature

Adding generation of signature files

tesonep committed:
Preparing release of v10.1.1

tesonep committed:
Merge pull request #752 from pharo-project/preparing-release-v10.1.1

Preparing release of v10.1.1

tesonep committed:
Fixing Release process

tesonep committed:
Fixing release

Rinzwind committed:
Merged commit 32b2be55c15f2e97 (latest ‘pharo10’).

Rinzwind committed:
Changed error checking macros used in the child process in ‘tty_spawn’ to use ‘_exit’ instead of ‘exit’, and ‘dprintf’ instead of ‘printf’, to avoid flushing the open streams copied, buffers included, from the parent process.

doste committed:
added a few test for primitiveFormat and the corresponding JIT method

doste committed:
Added a few tests

doste committed:
A little fix to the primitives that compare bytes. Now they can compare indexable objects byte by byte even if they are not indexable by bytes

guillep committed:
Fix VM build in MacOS sonoma

guillep committed:
Merge pull request #742 from Rinzwind/tty-spawn

Add library with function to spawn a process connected to a pseudo-terminal

guillep committed:
Merge pull request #758 from guillep/build-sonoma

Fix VM build in MacOS sonoma

doste committed:
Remove primitiveCompareBytes2 and keep primitiveCompareBytes

doste committed:
remove a test which was using a removed primitive

guillep committed:
Merge pull request #759 from doste/dojo-primitive-compare-bytes

Primitive to compare bytes

Commit:
Fix ##716. Translate picture to Englitsh

guillep committed:
Merge pull request #764 from Inao0/TranslateImageDoc

 Translate documentation picture to Englitsh

guillep committed:
Update CHANGELOG of version 10.2.0

guillep committed:
Update CMakeLists.txt to version 10.2.0

guillep committed:
Merge pull request #765 from guillep/prepare-10.2.0

Prepare Release 10.2.0

tesonep committed:
Moving pharo.signatures files for OSX Bundle to Resources folder

tesonep committed:
Adding the missing tty.c file in the packaging.

guillep committed:
Do not allow comparing objects of different types even though they are both bytes/words, failing the primitive.
This prevents wrongly comparing wide strings with byte strings on a byte-per-byte basis.

Co-authored-by: doste <ignbianchi@gmail.com>

guillep committed:
Merge pull request #771 from tesonep/adding-missing-file-to-package

Adding the missing tty.c file in the packaging.

guillep committed:
Merge pull request #772 from guillep/fix/comparing-different-formats

Do not allow comparing objects of different types

guillep committed:
Prepare release 10.2.1

guillep committed:
Merge pull request #773 from guillep/merge-10-2-1

Prepare release 10.2.1

guillep committed:
Bump to MacOS 11 and above

guillep committed:
Fix tautological-pointer-compare warnings

jordanmontt committed:
Removed Cogit>>#voidNSSendCache: method as it has no senders and it has a call to a non-existing method.

PalumboN committed:
Refactoring code to avoid warning

guillep committed:
Cast constant to 32 bit int to avoid warning

guillep committed:
Re-enable -Wno-incompatible-function-pointer-types

PalumboN committed:
Adding sqInt return type declaration on StackInterpreter>>#stackPointerIndexForFrame:

PalumboN committed:
Remove -Wno-conditional-type-mismatch flag

PalumboN committed:
Avoid no-conditional-type-mismatch on C code

jordanmontt committed:
Removed unused functions

doste committed:
a few fixes for warnings about return types

PalumboN committed:
Fix -Wno-incompatible-pointer-types warning

jordanmontt committed:
Removed not used method on the generated plugins

guillep committed:
Fix function pointer comparison with cast

jordanmontt committed:
Removed unused functions from LargeIntegers

guillep committed:
re enable warning incompatible-pointer-types

jordanmontt committed:
Remvoed unused functions from DSAPrims genereted plugin

PalumboN committed:
Fix (non)const incompatible types

guillep committed:
Fix shift-negative-value warning and reenable warning

guillep committed:
Do not use asserta: in a statement

guillep committed:
Remove unused value warnings from B2DPlugin

guillep committed:
Re-enable warnings

guillep committed:
Merge pull request #777 from guillep/constant-conversion-warning

Constant conversion warning

guillep committed:
Merge pull request #775 from guillep/warnings-osprocess

Fix tautological-pointer-compare warnings

guillep committed:
Merge pull request #776 from pharo-project/removing-voidNSSendCache

Removed Cogit>>#voidNSSendCache:

guillep committed:
Merge pull request #778 from guillep/incompatible-function-pointer-types-warning

Re-enable incompatible-function-pointer-types warning

guillep committed:
Merge pull request #780 from pharo-project/unused-functions-warning

Removed unused functions from the C written by hand

guillep committed:
Merge branch 'pharo-10' into fix-warning-return-type

guillep committed:
Merge pull request #783 from guillep/warning-function-types

Fix function pointer comparison warning

guillep committed:
Merge branch 'pharo-10' into warning-shift-negative-value

guillep committed:
Merge branch 'pharo-10' into remove-warnings-c-compiler

jordanmontt committed:
Removed interruptPending instance variable from the StackInterpreter.

guillep committed:
Merge pull request #781 from pharo-project/fix-warning-return-type

Fix -return-type warnings

guillep committed:
Merge pull request #785 from guillep/warning-shift-negative-value

Fix shift-negative-value warning and reenable warning

guillep committed:
Merge branch 'pharo-10' into remove-warnings-c-compiler

guillep committed:
Merge pull request #789 from PalumboN/remove-warnings-c-compiler

Remove warnings c compiler

guillep committed:
Merge pull request #793 from pharo-project/stack-interpreter-interrump-pending

Remove StackInterpreter interruptPending instance variable

guillep committed:
Merge branch 'pharo-10' into warning-asserta

guillep committed:
Merge pull request #788 from guillep/warning-asserta

Do not use asserta: in a statement

guillep committed:
Use cmake support for MacOS compilation CMAKE_OSX_DEPLOYMENT_TARGET instead of manual flags

guillep committed:
Merge branch 'pharo-10' into bump-macos

guillep committed:
Do not set CMAKE_OSX_DEPLOYMENT_TARGET as CACHE

guillep committed:
Merge branch 'bump-macos' of github.com:guillep/pharo-vm into bump-macos

guillep committed:
Merge pull request #792 from guillep/bump-macos

Bump to MacOS 11 and above

guillep committed:
Merge pull request #770 from tesonep/move-pharo.signatures-inOSX

Moving pharo.signatures files for OSX Bundle to Resources folder

RenaudFondeur committed:
Do not retain any selectors in CogAbstractInstruction

RenaudFondeur committed:
Remove overly conservative requiredMethodNames:

RenaudFondeur committed:
remove warning because of type and unsigned shift issues

guillep committed:
Merge pull request #798 from RenaudFondeur/warning-unused-functions

Do not retain any selectors in CogAbstractInstruction

tesonep committed:
Adding an implementation of the aio.c using EPOLL in Linux.
Avoiding the limit of file descriptors

RenaudFondeur committed:
add tests for type harmonization

guillep committed:
Merge pull request #800 from RenaudFondeur/fixWarning

remove warning because of type and unsigned shift issues

guillep committed:
Merge branch 'pharo-12' into cleanup/frame-unification

RenaudFondeur committed:
Merge branch 'pharo-project:pharo-10' into Harmonize

guillep committed:
Merge pull request #807 from RenaudFondeur/Harmonize

add tests for type harmonization

doste committed:
Improved the primitiveFormat, fix the jump target. And improved the concretizeMoveMbrR method,  it doesn't need to move a 0 because it concretizes to the 'LDURB' instruction which zero-extends the value

guillep committed:
Merge pull request #802 from doste/primitiveFormat

Primitive format

tesonep committed:
Fixing warnings that are errors in newer versions of clang

tesonep committed:
Improving the EPoll implementation: creating the descriptor each time and removing select from socket writable

tesonep committed:
Merge branch 'fix/errors-in-new-clang' into improve/using-epoll-for-linux

tesonep committed:
Adding new implementation of socketWritable for OSX and Windows

tesonep committed:
Fixing typo

tesonep committed:
In Windows, we need to do the call and handle the WSAEWOULDBLOCK error. So we assume, we can always write

tesonep committed:
We can always assume we are not going to block and use the events mechanisms

tesonep committed:
Removing socketWritable check as we can always use the event system to handle the would block error in send/sendto

tesonep committed:
Merge pull request #805 from tesonep/improve/using-epoll-for-linux

Adding an implementation of the aio.c using EPOLL in Linux.

tesonep committed:
Preparing for v10.2.1 release

tesonep committed:
Adding macro for win32. Fix #753

tesonep committed:
Fixing changelog

Rinzwind committed:
Fixed bug in ‘sqAcceptSSL’ in ‘sqMacSSL.c’ which, after setting the state to ‘SQSSL_CONNECTED’, returned the value of ‘SQSSL_OK’ (zero) rather than the number of bytes written to the output buffer.

Rinzwind committed:
Made ‘sqSetupSSL’ in ‘sqMacSSL.c’ handle the ‘CERTNAME’ property having been set (by searching for a valid identity with the property value as its subject through ‘SecItemCopyMatching’ and setting it as the SSL session context’s certificate through ‘SSLSetCertificate’).

RenaudFondeur committed:
New harmonize rule (#817)

Simple new rule for harmonization

- change in the tests to adapt to new rule
- add some test for negative number
- test for a more non-destructive rule

RenaudFondeur committed:
New harmonize rule (#817)

Simple new rule for harmonization

- change in the tests to adapt to new rule
- add some test for negative number
- test for a more non-destructive rule

tesonep committed:
Ignoring EAGAIN in epoll_wait

tesonep committed:
Merge pull request #818 from tesonep/Ignoring-EAGAIN-in-epoll_wait

Ignoring EAGAIN in epoll_wait

RenaudFondeur committed:
changes in methods to fix a type conflict

RenaudFondeur committed:
change to primitiveMoveToPermSpaceInBulk

RenaudFondeur committed:
change to activateNewFullClosure: blockClosure method: theMethod numArgs: numArgs mayContextSwitch: mayContextSwitch

RenaudFondeur committed:
change to checkStackPointerIndexForFrame:

RenaudFondeur committed:
change to allocationGranularity

RenaudFondeur committed:
a first version of Slang with no type conflict and exception if one appear

RenaudFondeur committed:
change in baseline

RenaudFondeur committed:
first test of new comportement

RenaudFondeur committed:
advance in methods, problem with supression

RenaudFondeur committed:
advance cleaning ast

guillep committed:
Making it loadable in P12
 - update dependency to pharo-opal-simd-bytecode
 - update usage of undeclareds
 - update method protocols

guillep committed:
Making it loadable in P12
 - update dependency to pharo-opal-simd-bytecode
 - update usage of undeclareds
 - update method protocols

guillep committed:
Update build version to P12

guillep committed:
Merge pull request #816 from Rinzwind/squeakssl-macos-certname

Extend macOS implementation of SqueakSSL plugin to support setting a certificate on the SSL session context

guillep committed:
Merge pull request #814 from tesonep/fix/issue-753

Adding macro for win32.

guillep committed:
Update macOS VM hash

guillep committed:
Bump version to 10.3.0 in CMake

guillep committed:
Update CHANGELOG to version 10.3.0

guillep committed:
Update latest P12 image hash

guillep committed:
Define variables as expected in Pharo 12

guillep committed:
Remove usage of `CommandLineArguments class >> default` that does not exist anymore in Pharo 12

guillep committed:
Merge pull request #827 from guillep/prepare-release-10-3-0

Prepare release 10 3 0

guillep committed:
Update tests to Pharo 12

guillep committed:
Merge pull request #829 from guillep/redo-294

Redo 294

guillep committed:
Merge pull request #825 from guillep/update-dev-to-p12

Making it loadable in P12

guillep committed:
Hoist stack manipulation

guillep committed:
Merge remote-tracking branch 'pharo-project/pharo-10' into fix/pharo11998

guillep committed:
Merge pull request #520 from guillep/fix/pharo11998

Added test on extended store and pop

guillep committed:
Fix tests and update deprecations

guillep committed:
Merge branch 'pharo-10' into update-ci-to-p12

RenaudFondeur committed:
test for unused variables

RenaudFondeur committed:
small change

RenaudFondeur committed:
fix small change

RenaudFondeur committed:
small change 2

RenaudFondeur committed:
fix small change 2

guillep committed:
Update hash of windows and macOS-arm64 vms

guillep committed:
Fix Linux x64 hash

RenaudFondeur committed:
fix small change 2

guillep committed:
Add deprecated methods

guillep committed:
Merge bf886610371bf64a1e5d5977d3f6176ef5b957b8

guillep committed:
Update linux x64 vm checksum

RenaudFondeur committed:
add statement to the nodes to parse

RenaudFondeur committed:
continue adding switch support

RenaudFondeur committed:
maybe a fonctional process of removing unused variable

RenaudFondeur committed:
adding struct accessor in the clening process

guillep committed:
Fix test depending on Pharo methods (that were removed)

guillep committed:
Merge 9c8b167fd5c95f5d0d5f8b5da4d3ea7bd96414a6

guillep committed:
Fixed compatibility of mocks

guillep committed:
Fix deprecations

guillep committed:
Remove unused primitives causing warnings

guillep committed:
Merge pull request #826 from guillep/update-ci-to-p12

Update build version to P12

RenaudFondeur committed:
adavance in adding statmenent as expression

RenaudFondeur committed:
adavance in supporting inlining in return and assignement

RenaudFondeur committed:
fix a lot of unused expression

RenaudFondeur committed:
Merge 51626b1bf4a2dca376cc0ab553e90447128a7ef9

RenaudFondeur committed:
Merge remote-tracking branch 'origin/pharo-10' into fixUnusedExpression

tesonep committed:
Adding message when aborting

tesonep committed:
- Moving repeated simulator code to the Trait
- Adding an origin message for handling the error of growing old space when a limit arrives

RenaudFondeur committed:
advance in tests

RenaudFondeur committed:
remove self halt

guillep committed:
Merge pull request #833 from tesonep/improving-log-of-oldSpaceLimit

Improving log of old space limit error reporting

RenaudFondeur committed:
advance in tests

RenaudFondeur committed:
merge

RenaudFondeur committed:
remove unused class

RenaudFondeur committed:
better tests

RenaudFondeur committed:
Merge branch 'pharo-10' into fixUnusedExpression

guillep committed:
Merge 99d178d2db5c9544d383921e9bc7ff62ae4e848e

guillep committed:
Format

guillep committed:
format

RenaudFondeur committed:
Merge 0e9f1ce17785bdde02bacacc6656b68d00c5f7e6

RenaudFondeur committed:
first visitor

RenaudFondeur committed:
add a working visitor and a better class for tests

RenaudFondeur committed:
remove commented code, unused methods and rename cleanASTOf: to removeUnusedNodesInBranch: . 
supress harmonizeSignedAndUnsignedTypesIn: , it has no sender and look exactly like an old version of  harmonizeReturnTypesIn: now in CCodeGenerator

RenaudFondeur committed:
tests for TDD regarding conditional send

RenaudFondeur committed:
improve and add a lot of  tests. refactoring of the dead code cleaning process

RenaudFondeur committed:
add tests for switch, found a bug related to inlining

RenaudFondeur committed:
fix the bug related to inlining by setting parents in postCopy : 
when inlining happens, we produce a copy of the method's AST but we only ensure that the children are correct meaning we cannot iterate the AST from bottom top anymore.

RenaudFondeur committed:
small changes in protocols

RenaudFondeur committed:
change incorrect comment

RenaudFondeur committed:
cleaner code thamks to mutation testing

RenaudFondeur committed:
remove unused cast and expression

RenaudFondeur committed:
add more comment support to slang : ensure there is a begin and end comment to every method inlined and restore lost comment. Include small changes in a lot of method executed after inlinining to make it work because of end comment.

change test related to inlining to include comments in their result.

RenaudFondeur committed:
change for comment in expression

RenaudFondeur committed:
change to comment in statement expression

RenaudFondeur committed:
fix change in comment in expression

RenaudFondeur committed:
change in visitExpressionList to handle end comment.

change in existing test to show this behavior

RenaudFondeur committed:
use a CCommentNode instead of an CEmptyNode to handle comment

RenaudFondeur committed:
fix expressionList + add a test for comment in expressionList

RenaudFondeur committed:
fix simulationOnly:

RenaudFondeur committed:
fix bug with empty inlined method

RenaudFondeur committed:
forgot to commit a method in test

RenaudFondeur committed:
handle correctly comment series

RenaudFondeur committed:
small change

RenaudFondeur committed:
add support for empty inlined method in the end of the list

RenaudFondeur committed:
fix bug related to the build of stackVm

RenaudFondeur committed:
actual fix

RenaudFondeur committed:
start of better tests

RenaudFondeur committed:
fix a bug with empty method (having only comments) causing exitVar:label:  to  put statements between them

RenaudFondeur committed:
change the name and comment of test

RenaudFondeur committed:
change the name and comment of test

RenaudFondeur committed:
fix a bug related to empty method (having only comments) where exitVar:label: would place statements between them

RenaudFondeur committed:
actual fix

RenaudFondeur committed:
Merge branch 'fixBugBetterComment' into betterComment

guillep committed:
Update smalltalksrc/Slang/CSlangPrettyPrinter.class.st

RenaudFondeur committed:
prepare comment support

RenaudFondeur committed:
fix the 'warning, attempt to include <sys/stat.h> /* for e.g. mkdir */ a second time ' happening in Slang

guillep committed:
Merge pull request #838 from RenaudFondeur/betterComment

a better comment support for Slang

guillep committed:
Merge branch 'pharo-10' into fixUnusedExpression

guillep committed:
Merge pull request #819 from RenaudFondeur/FixMethodTypeConflict

a first version of Slang with no type conflict and an exception if one appear

guillep committed:
Merge pull request #837 from RenaudFondeur/fixNegativeShiftAndUnusedCCoerce

remove unused cast and expression

guillep committed:
Merge pull request #840 from RenaudFondeur/fixMultipleHeaderInclude

fix warnings related to multiple include of the same header file

guillep committed:
Merge pull request #832 from RenaudFondeur/fixUnusedExpression

Fix a lot of unused expression

RenaudFondeur committed:
add a comment explaining why declareCVarsIn: es empty in subclasses

RenaudFondeur committed:
Merge branch 'pharo-project:pharo-10' into addCommentForMultipleHeaderInclude

RenaudFondeur committed:
Merge branch 'pharo-10' into fixUnusedExpressionHandleComment

RenaudFondeur committed:
change in removeUnusedNodesInBranch: to handle comments

guillep committed:
Merge pull request #842 from RenaudFondeur/addCommentForMultipleHeaderInclude

add a comment explaining why declareCVarsIn: is empty in some subclasses

guillep committed:
Merge pull request #843 from RenaudFondeur/fixunusedExpressionHandleComment

small change in dead code elimination to considers a method with only comments empty

guillep committed:
Merge ece0d66c505168ff524370838763e2b322d683d3

guillep committed:
Finish merge with Pharo-10 branch

guillep committed:
Ban name rules

guillep committed:
Add missing instance variable

guillep committed:
Fix store trampoline test

guillep committed:
Restore statically-resolved selector renaming

guillep committed:
Fix simulation memory access tests

guillep committed:
Fix code generation and type conflicts

guillep committed:
Remove display functions

guillep committed:
Merge 357f978c06e8ba5c28f4dd545c904a5d1198b1dd

guillep committed:
Finish merge with frame cleanup branch

guillep commented:
I'll merge this into the experimental branch, some tests were broken, we will investigate further.

<img width="741" alt="imagen" src="https://github.com/user-attachments/assets/942f098e-12a4-4811-9a9b-0992fb9c369f">
